Content.Shared/Chemistry/Reaction/ChemicalReactionSystem.cs

```diff
@@ -249,30 +258,23 @@ private bool ProcessReactions(Solution solution, EntityUid owner, FixedPoint2 ma
             if (products == null)
                 return false;
 
-            // Remove any reactions that were not applicable. Avoids re-iterating over them in future.
-            foreach (var proto in toRemove)
-            {
-                reactions.Remove(proto);
-            }
```

Comment:
This implicitly assumed that the properties/contents of the solution haven't changed, which isn't generally true.